src/python/tests/test_db.py

```diff
@@ -0,0 +1,285 @@
+
+        Args:
+            src: Relative directory path where the test database schema and content files are located,
+                starting from ``ensembl-compara/src/python/tests/databases``.
+            name: Name to give to the new database (it will be prefixed by the username).
+            expectation: Context manager for the expected exception, i.e. the test will only pass if that
+                exception is raised. Use :class:`contextlib.nullcontext` if no exception is expected.
+
+        """
+        server_url = request.config.getoption('server')
+        with expectation:
+            db_name = name if name else src.name
+            self.dbs_created[db_name] = UnitTestDB(server_url, pytest.dbs_dir / src, name)
+            # Check that the database has been created correctly
+            assert self.dbs_created[db_name], "UnitTestDB should not be empty"
+            assert self.dbs_created[db_name].dbc, "UnitTestDB's database connection should not be empty"
+            database_name = os.environ['USER'] + '_' + db_name
+            if self.dbs_created[db_name].dbc.dialect == 'sqlite':
+                # Need to add the path to the database and ".db" extension
+                database_name = make_url(server_url).database + '/' + database_name + '.db'
+            assert self.dbs_created[db_name].dbc.db_name == database_name, \
+                f"Expected database name to be '{database_name}'"
+            # Check that the database has been loaded correctly from the dump files
+            result = self.dbs_created[db_name].dbc.execute("SELECT * FROM main_table")
+            assert len(result.fetchall()) == 10, "Unexpected number of rows found in 'main_table' table"
+
+    @pytest.mark.parametrize(
+        "db_name",
+        [
+            pytest.param('reference', marks=pytest.mark.dependency(depends=['reference'], scope='class')),
+            pytest.param('renamed', marks=pytest.mark.dependency(depends=['renamed'], scope='class')),
+        ],
+        ids=pytest.get_param_repr
+    )
+    def test_drop(self, db_name: str) -> None:
+        """Tests that the previously created object :class:`UnitTestDB` is dropped correctly.
+
+        Args:
+            db_name: Key assigned to the UnitTestDB created in :meth:`TestUnitTestDB.test_init()`.
+
+        """
+        self.dbs_created[db_name].drop()
+        if self.dbs_created[db_name].dbc.dialect == 'sqlite':
+            # For SQLite databases, just check if the database file still exists
+            assert not Path(self.dbs_created[db_name].dbc.db_name).exists(), \
+                "The database file has not been deleted"
+        else:
+            with raises(OperationalError, match=r'Unknown database'):
+                self.dbs_created[db_name].dbc.execute("SELECT * FROM main_table")
+
+
+@pytest.fixture(scope='class')
+def test_dbc(request: FixtureRequest, db_factory: Callable) -> None:
+    """Assigns a :class:`DBConnection` object to a pytest class attribute.
```

Comment:
```suggestion
    """Assigns a :class:`DBConnection` object to a pytest class attribute for use in further tests.
```
Maybe it's just me, but I'm finding this code really hard to pick apart. Some methods are for setup of objects, some run tests, some are framework related. For me, the doc strings aren't making the distinctions clear enough.

Reply:
It took me a while to get my head around how `pytest` works/needs things set up, and I completely understand your confusion. I'll have a look to see if the whole readability can be improved.